### PR TITLE
WPC-1047 feat(website-builder): author list

### PR DESCRIPTION
**Pull request by antkiewiczk**

### CODE REVIEW

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    onQuery?: (
+      queryResult: Pick<QueryResult<AuthorListQuery>, 'data' | 'loading' | 'error' | 'refetch'>
+    ) => void
+  } & BuilderAuthorListProps
+
+export function AuthorList({onQuery, className}: AuthorListProps) {
+  const {AuthorChip} = useWebsiteBuilder()
+
+  const authorListQueryVariables = {
+    // should the AuthorList accept variables?
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:27]: Yes, see event list for an example

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import {useAuthorListQuery, AuthorListQuery, FullAuthorFragment} from '@wepublish/website/api'
+import {styled, css} from '@mui/material'
+import {QueryResult} from '@apollo/client'
+import {useEffect} from 'react'
+import {BuilderAuthorListProps, useWebsiteBuilder} from '@wepublish/website/builder'
+
+export const AuthorTileWrapper = styled('a')`
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:7]: This causes a full page reload

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  color: inherit;
+`
+
+export const AuthorTileImageWrapper = styled('div')`
+  width: 240px;
+  height: 240px;
+`
+
+export const AuthorTileContentWrapper = styled('div')`
+  margin-top: 20px;
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:22]: Use gaps and not margin for layouting if possible.

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+`
+
+export const AuthorListWrapper = styled('article')`
+  display: grid;
+  grid-template-columns: repeat(4, minmax(230px, 1fr));
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:27]: This is not mobile friendly. Should prob also not have a hardcoded pixel value

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+const imageStyles = css`
+  border-radius: 50%;
+`
+
+export type AuthorListProps = Pick<BuilderAuthorListProps, 'authors'> & {
+  onQuery?: (
+    queryResult: Pick<QueryResult<AuthorListQuery>, 'data' | 'loading' | 'error' | 'refetch'>
+  ) => void
+  skip?: number
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:40]: Please check Event List how to pass down variables. This is missing things now and not future proof.

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  className?: string
+  author: FullAuthorFragment
+}
+
+export function AuthorList({onQuery, className, skip, take}: AuthorListProps) {
+  const {
+    elements: {Image, Paragraph, H6}
+  } = useWebsiteBuilder()
+
+  const AuthorTile = ({className, author}: AuthorTileProps) => (
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:54]: Please don't mix containers with view components. 
Components inside a component is also an antipattern as the reference is different on **every render**

**Review by Itrulia — CHANGES_REQUESTED** [review]
Missing Storybook, Tests, Website Builder implementation & Split off Container / View Component

File: libs/author/website/src/lib/author-list/author-list-item.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+export const AuthorListItemImageWrapper = styled('div')`
+  width: 240px;
+  height: 240px;
+`
+
+const wrapperStyles = (theme: Theme) => css`
+  display: grid;
+  gap: ${theme.spacing(3)};
+  grid-template-rows: 240px 1fr;
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list-item.tsx:13]: Any specific reason the child sets it's own (hardcoded) width?

File: libs/website/builder/src/lib/author.interface.ts [hunk 8]
```diff
@@ -12,6 +17,21 @@ export type BuilderAuthorChipProps = {
...
+  author: FullAuthorFragment
+  className?: string
+}
+
+export type BuilderAuthorListProps = Pick<
+  QueryResult<AuthorListQuery>,
+  'data' | 'loading' | 'error'
+> & {
+  className?: string
+  authors?: ComponentType<BuilderAuthorChipProps>
```
Comment (Itrulia) [inline libs/website/builder/src/lib/author.interface.ts:30]: Shouldn't be there. I guess c&p? :D

File: libs/author/website/src/lib/author-list/author-list-item.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  text-align: center;
+  text-decoration: none;
+  color: inherit;
+`
+
+const imageStyles = css`
+  border-radius: 50%;
+`
+
+type AuthorListItemProps = {
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list-item.tsx:25]: Use website builder props not these

**Review by Itrulia — CHANGES_REQUESTED** [review]
A few comments. Website example implementation is missing FYI

File: libs/website/api/src/lib/schemas/author.graphql [hunk 10]
```diff
@@ -21,3 +21,20 @@ query Author($id: ID, $slug: Slug) {
     ...FullAuthor
   }
 }
+
+query AuthorList($filter: String, $cursor: ID, $take: Int, $skip: Int) {
```
Comment (Itrulia) [inline libs/website/api/src/lib/schemas/author.graphql:25]: Oh just saw we do `$filter: String`, can we have the full filter list here?

Comment (antkiewiczk) [reply libs/website/api/src/lib/schemas/author.graphql:25]: doesn't it require changes in the API? and then also in other parts like editor?

File: libs/author/website/src/lib/author-list/author-list-item.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    elements: {Image, Paragraph, H6, Link}
+  } = useWebsiteBuilder()
+
+  return (
+    <Link css={wrapperStyles(theme)} className={className} href={author.slug}>
+      <AuthorListItemImageWrapper>
+        {author.image && <Image image={author.image} square css={imageStyles} />}
+      </AuthorListItemImageWrapper>
+
+      <div>
```
[the remark this answers is not in the document]
Comment (antkiewiczk) [reply libs/author/website/src/lib/author-list/author-list-item.tsx:42]: like this?
`export const AuthorListItemContentWrapper = styled('div')```

[the remark this answers is not in the document]
Comment (Itrulia) [reply libs/author/website/src/lib/author-list/author-list-item.tsx:42]: Or `AuthorListItemContent` thanks!

File: libs/website/api/src/lib/schemas/author.graphql (hunk 10 above)
Comment (Itrulia) [reply libs/website/api/src/lib/schemas/author.graphql:25]: Nah, this is just how you define it locally, below it's applied partially.

String -> AuthorFilter

and then below 

filter: {name: $filter} -> filter: $filter

File: libs/author/website/src/lib/author-list/author-list-item.tsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+  width: 240px;
+  height: 240px;
+`
+
+export const AuthorListItemContent = styled('div')``
+
+const wrapperStyles = (theme: Theme) => css`
+  display: grid;
+  gap: ${theme.spacing(3)};
+  width: 240px;
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list-item.tsx:15]: Should come from parent

File: libs/author/website/src/lib/author-list/author-list-item.tsx [hunk 13]
```diff
@@ -0,0 +1,51 @@
+import {Author} from '@wepublish/website/api'
+import {Theme, styled, css, useTheme} from '@mui/material'
+import {useWebsiteBuilder} from '@wepublish/website/builder'
+
+export const AuthorListItemImageWrapper = styled('div')`
+  width: 240px;
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list-item.tsx:6]: should come from parent

**Review by Itrulia — COMMENTED** [review]
Looks good, just 1 small thing about the child setting the width instead of the parent.

File: libs/author/website/src/lib/author-list/author-list.tsx [hunk 14]
```diff
@@ -0,0 +1,21 @@
+import {styled} from '@mui/material'
+import {BuilderAuthorListProps, useWebsiteBuilder} from '@wepublish/website/builder'
+
+export const AuthorListWrapper = styled('article')`
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(240px, 1fr));
+  gap: ${({theme}) => theme.spacing(4)};
+  justify-items: center;
```
Comment (Itrulia) [inline libs/author/website/src/lib/author-list/author-list.tsx:8]: prob want to go from left to right?

**Review by Itrulia — APPROVED** [review]

**Pull request merged**